### ISSUE TITLE
When doing index rebuilds in on partitioned tables, trying to do by @PartitionLevel = 'N' generates error

### ISSUE THREAD

**Issue by SQLKirby** [olahallengren/sql-server-maintenance-solution#307]
When doing index rebuilds in on partitioned tables, trying to do by @PartitionLevel = 'N' generates error.
I have large databases with many partitioned tables.  I would like the ability to try to rebuild indexes using the Resumable option but not by partition.

**SQL Server version and edition**
Microsoft SQL Server 2017 (RTM-CU13) (KB4466404) - 14.0.3048.4 (X64) 
	Nov 30 2018 12:57:58 
	Copyright (C) 2017 Microsoft Corporation
	Enterprise Edition (64-bit) on Windows Server 2016 Datacenter 10.0 <X64> (Build 14393: )`

**Version of the script**
Version: 2018-06-27 20:44:12 

**What command are you executing?**
	EXECUTE @RC = [OLA].[IndexOptimize] 
	   @Databases = @dblist
	  ,@FragmentationMedium = @FragmentationMedium
	  ,@FragmentationHigh = @FragmentationHigh
	  ,@FragmentationLevel1 = @FragmentationLevel1
	  ,@FragmentationLevel2 = @FragmentationLevel2
	  ,@MinNumberOfPages = @MinNumberOfPages
	  ,@MaxNumberOfPages = @MaxNumberOfPages
	  ,@SortInTempdb = @SortInTempdb
	  ,@PartitionLevel = 'N'
	  ,@MaxDOP = @MaxDOP
	  ,@FillFactor = @FillFactor
	  ,@LOBCompaction = @LOBCompaction
	  ,@Indexes = @Indexes
	  ,@Delay = @Delay
	  ,@WaitAtLowPriorityMaxDuration = @WaitAtLowPriorityMaxDuration
	  ,@WaitAtLowPriorityAbortAfterWait = @WaitAtLowPriorityAbortAfterWait
	  ,@DatabasesInParallel = 'N'
	  ,@Resumable = 'Y'

	  ,@LogToTable = @LogToTable
	  ,@Execute = @Execute
Note @Indexes = ALL_INDEXES, -PhoneHomeParsed.dbo.diskmanager_allocation, -PhoneHomeParsed.dbo.diskmanager_disk, -PhoneHomeParsed.dbo.iocommon_actioninfostate, -

**What output are you getting?**
Msg 4104, Level 16, State 1, Line 1
The multi-part identifier "partitions.partition_number" could not be bound.
Msg 50000, Level 16, State 1, Procedure OLA.IndexOptimize, Line 1523 [Batch Start Line 0]
The following objects in the @Indexes parameter do not exist: [PhoneHomeParsed].[dbo].[diskmanager_allocation], [PhoneHomeParsed].[dbo].[diskmanager_disk], [PhoneHomeParsed].[dbo].[iocommon_actioninfostate], [PhoneHomeParsed].[dbo].[raid_extent], [PhoneHomeParsed].[dbo].[scsiinitiator_devices], [PhoneHomeParsed].[dbo].[scsiinitiator_dsclunids], [PhoneHomeParsed].[dbo].[scsiinitiator_dscluns], [PhoneHomeParsed].[dbo].[scsitarget_mapzones], [PhoneHomeParsed].[dbo].[snapshot_actioninfostate], [PhoneHomeParsed].[dbo].[snapshot_dpmdedupmaptable], [PhoneHomeParsed].[dbo].[snapshot_tracepage], [PhoneHomeParsed].[dbo].[system_majoreventlog], [PhoneHomeParsed].[dbo].[utility_memsmallblock].
 ** the error seems to be on all the excluded tables.

**Comment by AndreasEDA** [comment]
I have the same Issue. Are there any solutions?

**Comment by SQLKirby** [comment]
Yes I fixed it in the proc for myself.  I sent the code to Ola, not sure if he implemented it yet.  If needed I can share again.

**Comment by AndreasEDA** [comment]
Hi

Thank’s. That would be great if you could share it with me

Thanks’ a lot

Best Regards
Andy

**Comment by SQLKirby** [comment]
Hi Andy:

So I'm sending you my fixed version of the IndexOptimize procedure. There
are notes in there on what I changed with my initials nearby KR.
Note the version of Ola's scripts I have is from Jan 14 of this year.  I
also make some other changes like install the scripts in the DBADB database
in the OLA schema.  So if you want to use my version you might have to make
some minor changes.
Otherwise you might just download the latest and apply the changes.
It seems that there aren't many of us out here using partitioning.  We are
heavily invested - I have hundreds of partitioned tables with many TB of
data!
Feel free to connect on linked in.
Good luck.

Kirby Richter
Database Architect/ Administrator
Dell EMC
sqlKirby@gmail.com

**Comment by pablomatto** [comment]
Hello. That happens to me too. could you share it with me? Thanks

**Comment by olahallengren** [comment]
This issue has been fixed in https://github.com/olahallengren/sql-server-maintenance-solution/pull/869. Thank you for reporting it.
